dispatches/models/fossil_case/properties/hitecsalt_properties.py

```diff
@@ -0,0 +1,424 @@
+                               'mass': pyunits.kg,
+                               'amount': pyunits.mol,
+                               'temperature': pyunits.K})
+
+class _StateBlock(StateBlock):
+    """
+    This Class contains methods which should be applied to Property Blocks as a
+    whole, rather than individual elements of indexed Property Blocks.
+    """
+    def initialize(blk, state_args=None,
+                   outlvl=0, hold_state=False,
+                   state_vars_fixed=False,
+                   solver=None,
+                   optarg={}):
+        """
+        Declare initialisation routine.
+        Keyword Arguments:
+            state_args = to be used if state block initialized independent of
+                         control volume initialize
+            outlvl : sets output level of initialisation routine
+                     * 0 = no output (default)
+                     * 1 = return solver state for each step in routine
+                     * 2 = include solver output infomation (tee=True)
+            optarg : solver options dictionary object (default=None)
+            solver : str indicating whcih solver to use during
+                     initialization (default = 'ipopt')
+            state_vars_fixed: Flag to denote if state vars have already been
+                              fixed.
+                              - True - states have already been fixed by the
+                                       control volume 1D. Control volume 0D
+                                       does not fix the state vars, so will
+                                       be False if this state block is used
+                                       with 0D blocks.
+                             - False - states have not been fixed. The state
+                                       block will deal with fixing/unfixing.
+            hold_state : flag indicating whether the initialization routine
+                         should unfix any state variables fixed during
+                         initialization (default=False).
+                         - True - states varaibles are not unfixed, and
+                                 a dict of returned containing flags for
+                                 which states were fixed during
+                                 initialization.
+                        - False - state variables are unfixed after
+                                 initialization by calling the
+                                 relase_state method
+        Returns:
+            If hold_states is True, returns a dict containing flags for
+            which states were fixed during initialization.
+        """
+        # Fix state variables if not already fixed by the control volume block
+        if state_vars_fixed is False:
+            # Fix state variables if not already fixed
+            Fflag = {}
+            Tflag = {}
+            Pflag = {}
+            
+            for k in blk.keys():
```

Comment:
You should make use of the IDAES `fix_state_vars` utility method for this.

Reply:
addressed